python/ql/src/Statements/ReturnOrYieldOutsideOfFunction.qhelp

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>In Python, <code>return</code> and <code>yield</code> statements as well as <code>yield from</code>
+expression can only be used within a function. Having them placed outside of a function or a class
+method will raise a <code>SyntaxError</code> at runtime.</p>
+</overview>
+
+<recommendation>
+<p>The presence of <code>return</code> and <code>yield</code> statements or <code>yield from</code>
+expression outside of a function or a class method suggests a logical error, so it is not possible
+to suggest a general fix.</p>
+</recommendation>
+
+<example>
+<p>In this example, a <code>return</code> statement is used outside of a class method in a class and
+a <code>yield</code> statement is used outside of a function in a scope of a module.</p>
+<sample src="ReturnOrYieldOutsideOfFunction.py" />
```

Comment:
Do you mean to add comments to the `.py` file and not in the `.qhelp` file?

Reply:
Normally any explanation goes in the qhelp file. I know that you said that an overall recommendation wasn't possible for alerts for this query, but perhaps there are some suggestions that would make sense for these

Reply:
I hear you. What do you say about 

The `return` statement could be placed into a class method or the class could be refactored into a function instead. The `yield` statement could be a part of a new generator function containing the `for` loop.

Reply:
Thanks. I can't comment on the validity of the Python, but assuming that's fine that LGTM.

It might be worth amending slightly to be explicit that these actions would fix the syntax error, perhaps: "The invalid class could be corrected by placing the `return` statement in a class method, or by refactoring the class into a function. The invalid `yield` statement could become part of a new generator function containing the `for` loop."

Reply:
Thank you for helping me to word this, this is perfect.